assets/src/edit-story/elements/video/frame.js

```diff
@@ -14,14 +14,127 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { CSSTransition } from 'react-transition-group';
+
 /**
  * Internal dependencies
  */
 import StoryPropTypes from '../../types';
 import MediaFrame from '../media/frame';
+import { elementFillContent } from '../shared';
+import { useStory } from '../../app/story';
+import { ReactComponent as PlayIcon } from '../../icons/play.svg';
+import { ReactComponent as PauseIcon } from '../../icons/pause.svg';
+
+const Play = styled(PlayIcon)`
+  width: 48px;
+  height: 48px;
+`;
+const Pause = styled(PauseIcon)`
+  width: 48px;
+  height: 48px;
+`;
+
+const Wrapper = styled.div`
+  ${elementFillContent}
+`;
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  cursor: pointer;
+
+  opacity: 0;
+  &.button-enter {
+    opacity: 0;
+  }
+  &.button-enter-active,
+  &.button-enter-done {
+    opacity: 1;
+    transition: opacity 200ms;
+  }
+  &.button-exit {
+    opacity: 1;
+  }
+  &.button-exit-active,
+  &.button-exit-done {
+    opacity: 0;
+    transition: opacity 200ms;
+  }
+`;
 
 function VideoFrame({ element }) {
-  return <MediaFrame element={element} />;
+  const { id } = element;
+  const [hovering, setHovering] = useState(false);
+  const videoNode = document.getElementById(`video-${id}`);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const {
+    state: { selectedElementIds },
+  } = useStory();
+  const isElementSelected = selectedElementIds.includes(id);
+  const noElementsSelected = !selectedElementIds.length;
+
+  const onPointerEnter = () => {
+    // Sync UI for auto-play on insert.
+    const currentlyPlaying = videoNode && !videoNode.paused;
+    if (currentlyPlaying && !isPlaying) {
+      setIsPlaying(true);
+    }
+    setHovering(true);
+  };
+
+  useEffect(() => {
+    if ((!isElementSelected || noElementsSelected) && videoNode) {
+      videoNode.pause();
+      videoNode.currentTime = 0;
+      setIsPlaying(false);
+    }
+  }, [isElementSelected, noElementsSelected, videoNode]);
+
+  const handlePlayPause = () => {
+    if (isPlaying) {
+      videoNode.pause();
+      setIsPlaying(false);
+    } else {
+      videoNode.play().then(() => {
+        setIsPlaying(true);
+      });
```

Comment:
Should we add a catch here and set the isplaying to `false`?

Reply:
`isPlaying` is guaranteed to be false here, and it will turn true only on play success, so unless we want to display a message - the catch is not needed.